sveltekit-authentication/index.md

```diff
@@ -0,0 +1,449 @@
+    }
+
+    if (event.route.id && shouldProtectRoute(event.route.id) && !event.locals.user) {
+        redirect(302,`/signin?redirectedFrom=${encodeURIComponent(url.pathname)}`)
+    }
+
+    // this is needed so that the response headers from SvelteKit do include the correct header to allow the browser to parse json requests
+    return await resolve(event, {
+        filterSerializedResponseHeaders: (key, value) => {
+            return key.toLowerCase() === 'content-type'
+        }
+    });
+}
+```
+
+Every single request goes pass trough this file. It checks the tokens and if valid sets the user id and the token to the locals object, which can be accessed througout SvelteKits Load Functions. If the Access Token is expired, a new one will be generated. Finally, we can define routes under the `(protected)` Directory. The "protected" keyword will not appear in the url, but is only visible in our file structure.
+
+In order for the authentication to work, however, we also need to tell SvelteKit to actually pass those local variables through to every other load function.
+
+Create a file `/+layout.server.js`:
+
+```js [/+layout.server.js]
+/** @type {import('./$types').LayoutServerLoad} */
+export async function load({locals}) {
+	return {
+		user: locals.user,
+		token: locals.token
+	};
+}
+```
+
+## Create Profile Page
+
+The locals will now be used in every `+page.js` file to initialize the Directus SDK with the user's session token. To test this out, let's now create our `profile` page, which is only accessible if the user is logged in.
+
+Create a file `/(protected)/profile/+page.js`
+
+```js [/(protected)/profile/+page.js]
+import getDirectusInstance from "$lib/directus";
+import { readMe } from "@directus/sdk";
+import { error } from "@sveltejs/kit";
+export async function load({ parent, fetch }) {
+  const { token } = await parent();
+  const directus = getDirectusInstance(fetch, token);
+  try {
+    return {
+      user: await directus.request(
+        readMe({
+          fields: ["*"],
+        })
+      ),
+    };
+  } catch (err) {
+    error(404, "User not found");
+  }
+}
+
+```
+
+As you see, we are getting the token and initialize our Directus Instance as usual. This time however, we also give it the Access Token so that every request will now have the User's Session attached. In this case, we are reading the user's profile. If you try this without the token, the request will fail because of missing permissions.
+
+Let's continue writing the Html Template:
+
+```svelte [/(protected)/profile/+page.svelte]
+<script>
+  export let data;
+</script>
+  
+<p>This is only visible if you are logged in</p>
+
+<p>Your Email: {data.user.email}</p>
+```
+
+You should now see a simple profile page if you are logged in under http://localhost:5173/profile
+
+![A simple example of a profile page showing the users email](profile1.webp)
+
+
+### Authenticated requests on the client and logging the user out
```

Comment:
Split into two sections

Reply:
did you mean to use the `code group`markdown component so that the js and svelte file are shown side-by-side ? if not please explain

Reply:
sry i think i understand now, you mean logout as its own section. ok sure will do